src/simulators.rs

```diff
@@ -124,34 +128,38 @@ impl<G: Generator> Server<G> {
         }
     }
 
-    // See comment above Client.tick for the use of the tick pattern. The return value here, if not
-    // None, is the packet that has finished being processed.
-    //
-    // The server operates via polling essentially, we process an entire packet each time the
-    // ticker returns to zero, if any. If the queue is empty this means the server is idle for that
-    // time duration and is recorded so internally.
-    pub fn tick(&mut self) -> Option<Packet> {
-        if self.ticker == 0 {
-            self.ticker = self.generator.next_event(self.resolution);
-            let packet = self.queue.pop_front();
-            match packet {
-                Some(_) => self.statistics.packets_processed += 1,
-                None => self.statistics.idle_count += 1,
+    // Server.process checks to see if a packet is currently being processed, and if so,
+    // increments Server.bits_processed, and if the resulting sum is equal to the bits
+    // in the packet, then it returns the packet and resets the state of Server.
+
+    pub fn process(&mut self) -> Option<Packet> {
+        match self.curr_packet.clone() {
+            Some(p) => {
+                self.bits_processed += self.pspeed / self.resolution;
+                if self.bits_processed as u32 == p.length {
```

Comment:
`self.bits_processed as u32 > p.length`

Reply:
And if it is greater than, pick up the next packet immediately if any and update the `bits_processed` counter accordingly.